tools/storybook-react/diagnose.sh

```
@@ -0,0 +1,175 @@
+  lsof -ti ":${PORT}" -sTCP:LISTEN 2>/dev/null | head -1
+}
+
+# Total CPU% across the process's threads; a wedged server sits at ~100 on one core.
+cpu_of() {
+  ps -o %cpu= -p "$1" 2>/dev/null | tr -d ' '
+}
+
+capture() {
+  local pid="$1" reason="$2"
+  mkdir -p "${OUT_DIR}"
+  local out="${OUT_DIR}/storybook-diagnosis-$(date +%Y%m%d-%H%M%S).txt"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)